.github/workflows/csound_builds.yml

```diff
@@ -124,6 +124,119 @@ jobs:
+    name: MacOS build (osxcross)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+          submodules: true
+
+      - name: Bootstrap VCPKG
+        run: ./vcpkg/bootstrap-vcpkg.sh
+
+      - name: Setup NuGet credentials
+        shell: bash
+        run: |
+          mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+          sources add \
+          -source "https://nuget.pkg.github.com/csound/index.json" \
+          -storepasswordincleartext \
+          -name "GitHub" \
+          -username "csound" \
+          -password "${{ secrets.GITHUB_TOKEN }}"
+          mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+            setapikey "${{ secrets.GITHUB_TOKEN }}" \
+            -source "https://nuget.pkg.github.com/csound/index.json"
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install build-essential liblzma-dev libxml2-dev libz-dev libbz2-dev cpio ninja-build
+
+      - name: Cache osxcross
+        id: osxcross
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/work/csound/csound/osxcross
+          key: ${{ runner.os }}-osxcross
+  
+      - name: Build osxcross
+        if: steps.osxcross.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/tpoechtrager/osxcross.git
+          sudo /home/runner/work/csound/csound/osxcross/tools/get_dependencies.sh
+          OSX_SDK="MacOSX14.0.sdk"
+          OSX_SDK_URL="https://github.com/joseluisq/macosx-sdks/releases/download/14.0/${OSX_SDK}.tar.xz"
+          curl -sSL "$OSX_SDK_URL" -o "/home/runner/work/csound/csound/osxcross/tarballs/$OSX_SDK.tar.xz"
+          UNATTENDED=1 /home/runner/work/csound/csound/osxcross/build_clang.sh
+          UNATTENDED=1 /home/runner/work/csound/csound/osxcross/build.sh
+
+      - name: Configure Build
+        run: |
+          $(eval /home/runner/work/csound/csound/osxcross/tools/osxcross_conf.sh)
+          for ARCH in x86_64 arm64; do
+              cmake -B build/$ARCH -S . -DUSE_VCPKG=1 -DCUSTOM_CMAKE=./platform/osxcross/custom-osx.cmake \
+                  -DCMAKE_SYSTEM_NAME=Darwin \
+                  -DOSXCROSS_TARGET_DIR=${OSXCROSS_TARGET_DIR} \
+                  -DOSXCROSS_SDK=${OSXCROSS_SDK} \
+                  -DOSXCROSS_TARGET=${OSXCROSS_TARGET} \
+                  -DCMAKE_OSX_ARCHITECTURES=${ARCH}
+          done
+
+      - name: Build Csound
+        run: |
+          $(eval /home/runner/work/csound/csound/osxcross/tools/osxcross_conf.sh)
+          for ARCH in x86_64 arm64; do
+            cmake --build build/$ARCH --config Release
+          done
+          mkdir artifact
+          zip -r artifact/build-$GITHUB_SHA.zip build
+
+      - name: Temporarily save osxcross build
+        uses: actions/upload-artifact@v2
+        with:
+          name: osxcross-artifact
+          path: artifact
+          retention-days: 1
```

Comment:
Not sure about this. We are hitting some usage limits and this will add to it. Why can this be the same job in the workflow to build and test?

Reply:
The cross-compiled artifacts are MacOS binaries and it's not possible to run them on the same Linux box that built them.

Reply:
I removed the tests workflow for osxcross to avoid creating the artifacts.